ddtrace/tracer/civisibility_transport_test.go

```diff
@@ -87,7 +88,9 @@ func runTransportTest(t *testing.T, agentless bool) {
 	if agentless {
 		t.Setenv(constants.CIVisibilityAgentlessEnabledEnvironmentVariable, "1")
 		t.Setenv(constants.CIVisibilityAgentlessURLEnvironmentVariable, srv.URL)
-		t.Setenv(constants.APIKeyEnvironmentVariable, "12345")
+		if shouldSetAPIKey {
+			t.Setenv(constants.APIKeyEnvironmentVariable, "12345")
+		}
```

Comment:
How does this change the outcome of the test?

Reply:
The test passes and all asserts are true. If I add `os.Exit` back to the code, the test fails.